common/datablocks/src/kernels/data_block_group_by_hash.rs

```diff
@@ -167,6 +167,7 @@ impl HashMethodSerializer {
         keys: Vec<Vec<u8>>,
         group_fields: &[DataField],
     ) -> Result<Vec<ColumnRef>> {
+        debug_assert!(!keys.is_empty());
```

Comment:
How about the `debug_assert` performance affects?

Reply:
it's only enabled in debug mode, which will be useful to detect the bugs in ci.

See: https://doc.rust-lang.org/std/macro.debug_assert.html